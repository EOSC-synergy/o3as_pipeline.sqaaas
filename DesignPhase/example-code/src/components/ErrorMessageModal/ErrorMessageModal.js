import React, { useState } from "react";

/**
 * displays an error message
 * @param {*} props 
 *  props.open -> whether the error message modal should be displayed
 *  props.onClose -> handles closing of the modal
 *  props.message -> error message
 * @returns a jsx file containing a modal with the given error message
 */
function ErrorMessageModal(props) {
    
    let i = props.open;
    i = props.onClose;
    i = props.message;
    

    return <div>
    </div>;
}

export default ErrorMessageModal;