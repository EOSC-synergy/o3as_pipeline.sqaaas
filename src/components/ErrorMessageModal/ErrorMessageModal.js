import React from "react";
import {useTheme} from "@mui/material/styles";
import {Button, Card, Grid, Modal, Typography} from "@mui/material";
import { IoCloseCircleOutline } from "react-icons/io5";

/**
 * Displays an error message.
 * @param {Object} props 
 * @param {boolean} props.isOpen -> whether the error message modal should be displayed
 * @param {function} props.onClose -> handles closing of the modal
 * @param {String} props.message -> error message
 * @returns {JSX.Element} a jsx file containing a modal with the given error message
 */
export default function ErrorMessageModal(props) {

    /**
     * The style of the modal.
     *
     * @type {{boxShadow: number, transform: string, top: string, bgColor: *, left: string, minWidth: string, position: string}}
     */
    const style = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        bgColor: useTheme().palette.grey[200],
        boxShadow: 24,
    };

    return (
        <>
            <Modal
                open={props.isOpen}
                onClose={props.onClose}
            >
                <Card sx={style}>
                    <Grid
                        container
                        direction="column"
                    >
                        <Grid
                            item
                            key="header"
                            sx={{backgroundColor:"#e85e6c", display: "flex", justifyContent: "center", py: 5}}
                        >
                            <IoCloseCircleOutline color="white" size={100} />
                        </Grid>
                        <Grid
                            item
                            key="message"
                            sx={{px: 5, pt: 5}}>
                            <Typography variant="h3" sx={{textAlign: "center"}}>Ooops!</Typography>
                            <br />
                            <Typography>{props.message}</Typography>
                        </Grid>
                        <Grid
                            item
                            key="button"
                            sx={{px: 5, pb: 5, display: "flex", justifyContent: "right"}}
                        >
                            <Button onClick={props.onClose}>Dismiss</Button>
                        </Grid>
                    </Grid>
                </Card>
            </Modal>
        </>
    );
}

// only needed for testing
ErrorMessageModal.defaultProps = {
    isOpen: false,
}