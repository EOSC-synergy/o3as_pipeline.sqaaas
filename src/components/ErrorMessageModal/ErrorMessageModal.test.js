import React from 'react';
import ReactDOM from 'react-dom';
import ErrorMessageModal from './ErrorMessageModal';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {shallow, configure} from 'enzyme';

describe('ErrorMessageModal component', () => {
    configure({adapter: new Adapter()});
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ErrorMessageModal /* isOpen={true} */ />);
    });

    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(wrapper, div);
    });

    // Snapshot test
    it('should render correctly from config file', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
