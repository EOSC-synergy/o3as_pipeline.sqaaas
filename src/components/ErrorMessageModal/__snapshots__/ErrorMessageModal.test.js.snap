// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessageModal component should render correctly from config file 1`] = `null`;
